cmd/launcher/extension.go

```diff
@@ -97,25 +99,60 @@ func createExtensionRuntime(ctx context.Context, db *bbolt.DB, launcherClient se
 		kolidelog.WithKeyValue("osqlevel", "stdout"),
 	)
 
-	runner := runtime.LaunchUnstartedInstance(
+	runnerOptions := []runtime.OsqueryInstanceOption{
 		runtime.WithOsquerydBinary(opts.OsquerydPath),
 		runtime.WithRootDirectory(opts.RootDirectory),
-		runtime.WithConfigPluginFlag("kolide_grpc"),
-		runtime.WithLoggerPluginFlag("kolide_grpc"),
-		runtime.WithDistributedPluginFlag("kolide_grpc"),
-		runtime.WithOsqueryExtensionPlugins(
-			config.NewPlugin("kolide_grpc", ext.GenerateConfigs),
-			distributed.NewPlugin("kolide_grpc", ext.GetQueries, ext.WriteResults),
-			osquerylogger.NewPlugin("kolide_grpc", ext.LogString),
-		),
 		runtime.WithOsqueryExtensionPlugins(ktable.LauncherTables(db, opts)...),
 		runtime.WithStdout(osqueryStdoutLogger),
 		runtime.WithStderr(osqueryStderrLogger),
 		runtime.WithLogger(logger),
 		runtime.WithOsqueryVerbose(opts.OsqueryVerbose),
 		runtime.WithOsqueryFlags(opts.OsqueryFlags),
 		runtime.WithAugeasLensFunction(augeas.InstallLenses),
-	)
+	}
+
+	if opts.Transport == "osquery" {
```

Comment:
This function is already pretty large, can we start to break it up for readability? I think the the contents of of the individual if / else blocks here are a good placed to start.

Reply:
You're not wrong... As the leading comment says:
```
// TODO: the extension, runtime, and client are all kind of entangled
// here. Untangle the underlying libraries and separate into units
```

I can't quite decide if I want to bring cleanup into scope here. On the other hand, no time like the present.

Reply:
Well... I tried splitting that out. I'm not sure. There's a lot of wiring that needs to be threaded through. I think this all mostly points at some much deeper refactoring in the future